### PR TITLE
New checklist feature

### PR DESCRIPTION
**Pull request by gonzalovelasco**

### CODE REVIEW

File: lib/screens/note/components/CheckListField/check_list_field.dart [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+          itemBuilder: (context, index) {
+            if (index == widget.uncheckedItems.length) {
+              return Row(
+                children: [
+                  IconButton(
+                    icon: const Icon(Icons.add),
+                    onPressed: addItem,
+                  ),
+                  Expanded(
+                    child: TextField(
```
Comment (matthewwong525) [inline lib/screens/note/components/CheckListField/check_list_field.dart:144]: maxLines should be 1 and clicking enter should go to the next field, I tried clicking enter and it did this which should not be the expected behaviour
image

**Review by matthewwong525 — CHANGES_REQUESTED** [review]
also disable markdown preview when in this mode because that breaks the checklist. (i.e. hide this button:
image

**Comment by matthewwong525** [comment]
only open checklist view if checklist is empty (when clicking from mobile toolbar)

**Comment by matthewwong525** [comment]
Also, when you click the checklist is first populated it should have autofocus!
1. Click checklist button
2. Focus should be within the checklist textfield

**Comment by matthewwong525** [comment]
also, could you take a look into this bug:
Export-1692985030215

File: lib/screens/note/components/CheckListField/check_list_field.dart (hunk 1 above)
Comment (gonzalovelasco) [reply lib/screens/note/components/CheckListField/check_list_field.dart:145]: Fixed 1e3b16f59c1d52a48c7f99d13cadf63c0a728169

**Comment by gonzalovelasco** [comment]
> only open checklist view if checklist is empty (when clicking from mobile toolbar)

673f54fb0e9d3f600f2eebdd1b6a1c720628b05c

**Comment by gonzalovelasco** [comment]
> Also, when you click the checklist is first populated it should have autofocus!
> 
> 1. Click checklist button
> 2. Focus should be within the checklist textfield

4d6d1fd057f6a599802ed1076f7f772ee489db9a

**Review by matthewwong525 — CHANGES_REQUESTED** [review]
two things to change:
1. Typing in a new field should not open and close the keyboard
2. When i click next /enter on the keyboard it should go to the next text field. (similar to google keep mobile)
3. when i click backspace and checklistitem is not the first checklist textfield then it'll delete it (similar to google keep mobile)
4. when i click ba

**Comment by matthewwong525** [comment]
look at my changes and apart from that LGTM

**Review by matthewwong525 — APPROVED** [review]
LGTM

**Pull request merged**